File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    hashdict={}
    sumhash=0
    
    for i in participant:
        hashdict[hash(i)]=i
        sumhash+=hash(i)
    
    for j in completion:
        sumhash-=hash(j)
    return hashdict[sumhash]
        

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
class Solution {
    public String solution(String[] participant, String[] completion) {
        StringBuilder sb = new StringBuilder();
        Map<String,Integer> m = new HashMap<String,Integer>();
        for(String s : completion){
            if(m.get(s)!=null){
                m.put(s,m.get(s)+1);
            }else{
                m.put(s,1);
            }
        }
        for(String s: participant){
            if(m.get(s)==null || m.get(s) ==0){
                sb.append(s);
                break;
            }else{
                m.put(s,m.get(s)-1);
            }
        }
        return sb.toString();
    }
}